packages/dev/core/src/FlowGraph/flowGraphEngine.ts

```diff
@@ -0,0 +1,36 @@
+import type { Scene } from "../scene";
+import { FlowGraph } from "./flowGraph";
+import { FlowGraphEventCoordinator } from "./flowGraphEventCoordinator";
+
+export class IFlowGraphEngineConfiguration {
+    scene: Scene;
+}
+/**
+ * The FlowGraphEngine class holds all of the existing flow graphs and is responsible for creating new ones.
+ * It also handles communication between them through an Event Coordinator
+ */
+export class FlowGraphEngine {
+    private _eventCoordinator: FlowGraphEventCoordinator;
+    private _flowGraphs: FlowGraph[] = [];
+
+    constructor(private _config: IFlowGraphEngineConfiguration) {
+        this._eventCoordinator = new FlowGraphEventCoordinator();
+    }
+
+    /**
+     * Creates a new flow graph and adds it to the list of existing flow graphs
+     * @returns a new flow graph
+     */
+    createGraph(): FlowGraph {
+        const graph = new FlowGraph({ scene: this._config.scene, eventCoordinator: this._eventCoordinator });
+        this._flowGraphs.push(graph);
+        return graph;
+    }
+
+    /**
+     * Starts all graphs
+     */
+    start() {

```

Comment:
What happens if the engine is already started but a new graph was added? Will the user need to execute start again? Wouldn't that re-start all graphs? (will probably re-set their current state).

I assume have a state for the engine would be helpful to decide how to deal with a newly created graph.